crates/ra_db/src/input.rs

```diff
@@ -312,12 +312,12 @@ impl fmt::Display for Edition {
 }
 
 impl Env {
-    pub fn set(&mut self, env: &str, value: String) {
-        self.entries.insert(env.to_owned(), value);
+    pub fn set<K: AsRef<str>, V: AsRef<str>>(&mut self, env: K, value: V) {
```

Comment:
I kind of like the original signature more:

* for non library-API functions I generally perefer not ot use `AsRef` polymorhism, for better type inference and less monomorphisation
* original signature avoided extra cloning of value

Reply:
@matklad when do you think this is reasonable to expose `AsRef<str>` parameters in public API, just curious?

Reply:
I think "never" is the right answer to this question. std uses `AsRef<OsStr>` for `Path`, and that is so that more specific `&str` literal could be passed into a function. For `str`, there's really no more specific literal type we can accept.